handler/turtle.py

```
@@ -0,0 +1,274 @@
+            for f in files:
+                self.all_files[f] = os.path.join(root,f)
+        self.logger.info("Found {0} files".format(len(self.all_files)))
+
+        # Generate item-level metadata graphs and store as files
+        with open(self.metadata_file, 'r') as f:
+            self.logger.info('Parsing the master metadata graph')
+            g = Graph().parse(f, format="turtle")
+            # For each of the unique subjects in the graph
+            for subj_uri in set([i for i in g.subjects()]):
+                # Get the item identifier
+                itembase = os.path.basename(subj_uri)
+                # Create the path to the output file
+                outfile = os.path.join(self.metadata_path, itembase) + '.ttl'
+                # Create a graph of all triples with that subject
+                itemgraph = Graph()
+                itemgraph += g.triples((subj_uri, None, None))
+                # Serialize the graph to the path location
+                self.logger.info('Serializing graph {0}'.format(outfile))
+                itemgraph.serialize(destination=outfile, format="turtle")
+
+        # If available, read batch index from file
+        if os.path.isfile(self.index_file):
+            self.logger.info("Reading batch index from {0}".format(self.index_file))
+            with open(self.index_file, 'r') as infile:
+                self.items = yaml.load(infile)
+
+        # Otherwise, construct the index by reading graph files
+        else:
+            for f in os.listdir(self.metadata_path):
+                fullpath = os.path.join(self.metadata_path, f)
+                # skip files starting with dot
+                if f.startswith('.'):
+                    self.extra_files.append(fullpath)
+                    continue
+                else:
+                    try:
+                        id = os.path.basename(f).rstrip('.ttl')
+                        self.items[id] = self.create_item(fullpath)
+                    except:
```

Having this bare `except` here is simply swallowing any errors without logging any sort of explanation. It would be better to explicitly catch the anticipated exceptions (e.g., `KeyError` when `create_item` cannot find a file in the index) and issue a warning via the logger.